cmd/utils/flags.go

```diff
@@ -1591,8 +1591,8 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 	CheckExclusive(ctx, DeveloperFlag, DeveloperPoWFlag, MainnetFlag, RopstenFlag, RinkebyFlag, GoerliFlag, YoloV3Flag, ClassicFlag, KottiFlag, MordorFlag, MintMeFlag)
 	CheckExclusive(ctx, LightServeFlag, SyncModeFlag, "light")
 	CheckExclusive(ctx, DeveloperFlag, DeveloperPoWFlag, ExternalSignerFlag) // Can't use both ephemeral unlocked and external signer
-	CheckExclusive(ctx, GCModeFlag, "archive", TxLookupLimitFlag)
 	if ctx.GlobalString(GCModeFlag.Name) == "archive" && ctx.GlobalUint64(TxLookupLimitFlag.Name) != 0 {
+		CheckExclusive(ctx, GCModeFlag, "archive", TxLookupLimitFlag)
 		ctx.GlobalSet(TxLookupLimitFlag.Name, "0")
```

Comment:
This will never be triggered. I suggest keeping either the `CheckExclusive` or set of `TxLookupLimit = 0`. @meowsbits what do you think?

Reply:
My thinking is that if somebody isn't setting TxLookupLimit but is setting gcmode=archive, the behavior of setting TxLookupLimit to zero (away from the default) is reasonable.

If someone is explicitly setting TxLookupLimit to zero, throwing an error because gcmode=archive was going to set it to zero anyway doesn't make sense.

But if somebody is trying to set TxLookupLimit to a value that is neither the default nor the value that gcmode=archive will set it to, then they probably want to know that's not a support configuration.

Reply:
I am sorry for the confusion @AusIV. I wanted to select the next line 1596. I agree that your PR is needed. What I am questioning/suggesting is that we either use `CheckExclusive` which will throw an error or we remove it and leave the next line to set it to `0` and continue.

Reply:
The current default value is non-zero. If you leave the default but use the archive flag, this will set it to zero.

I believe both lines are useful.

Reply:
I like what AusIV has here; reusing `CheckExclusive` to error "normally" if the `TxLookupLimitFlag` value _isn't zero_ with `archive` mode (since archive mode necessitates/implies keeping all tx indexes).

Reply:
> The current default value is non-zero. If you leave the default but use the archive flag, this will set it to zero.
> 
> I believe both lines are useful.

Sorry guys, I forgot to reply. Indeed nice catch @AusIV. I checked and it works fine. I was looking at the code only without checking the default value.